CMakeLists.txt

```diff
@@ -41,11 +41,20 @@ include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 include(CMakeDependentOption)
 
-message(STATUS "pybind11 v${pybind11_VERSION} ${pybind11_VERSION_TYPE}")
+if(NOT pybind11_QUIETLY)
+  message(STATUS "pybind11 v${pybind11_VERSION} ${pybind11_VERSION_TYPE}")
+endif()
 
 # Check if pybind11 is being used directly or via add_subdirectory
 if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
   set(PYBIND11_MASTER_PROJECT ON)
+
+  if(OSX AND CMAKE_VERSION VERSION_LESS 3.7)
```

Comment:
Sure.

In general, Linux is the only place you find "pre-installed" CMake, and therefore old versions. CMake is often a few versions late on supporting macOS/Windows with new features (`COMPILE_LANGAUGE` for example, took something like 10-12 versions of CMake to be supported), and much more important to have a recent version for the critical bug fixes / better defaults / support for the latest platforms that CMake has. You can't use a CMake version older than MSVC version, for example.

Anyway, I'll add the specific version notes here. 3.8 on Windows is simply due to the fact I can't test MSVC 2015 on GHA, and 3.8 is the first version that supported MSVC 2017. 3.7 is due to some bug in CMake macOS support that got fixed in 3.7, but I'll have to check and see which one.

Reply:
The bug only affected the test collection, by the way, so I only applied these minimum version checks for building pybind11's own tests.

Reply:
I've made the Windows one a warning, since it's just untested, not known to be broken.

Reply:
Only the CATCH download is broken on macOS + old cmake (maybe just the most recent versions of macOS?), so make this a warning instead of an error. If you already have catch, it should actually work on 3.4+. I've also removed the explicit mentions in the docs, 3.4+ is used everywhere.